core/protos/src/uint128_ext.rs

```diff
@@ -28,3 +28,27 @@ impl From<u128> for uint128_proto::Uint128 {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::convert::TryInto;
+
+    #[test]
+    fn parse_u128() {
+        let x: u128 = 123_123_123_123_123_123_123_123;
+        let y: uint128_proto::Uint128 = x.into();
+        let z: u128 = y.try_into().unwrap();
+        assert_eq!(z, x);
+    }
+
+    #[test]
+    fn fail_parse_u128() {
+        let x = uint128_proto::Uint128 { number: vec![], ..Default::default() };
+        let y: Result<u128, _> = x.try_into();
+        assert!(y.is_err());
+        let x = uint128_proto::Uint128 { number: vec![0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], ..Default::default() };
```

Comment:
nit: specify number of zeros explicitly? https://stackoverflow.com/a/29530932/341267